src/d3a/d3a_core/simulation.py

```diff
@@ -265,10 +265,12 @@ def _update_and_send_results(self):
             self.file_stats_endpoint(self.area)
             return
 
-        results = self.endpoint_buffer.prepare_results_for_publish()
-        if results is None:
-            return
-        self.kafka_connection.publish(results, self._simulation_id)
+        # don't send results to kafka when running on cli even though kafka connection is enabled:
+        if not self._started_from_cli and self.kafka_connection.is_enabled():
```

Comment:
If you want to go this way, then you should add the self._started_from_cli to the should_export_stats method, in order to be applied for the whole simulation class and not just here.

Reply:
OK, great! Will change that.

Reply:
Actually, now that I tried it again (had before but forgot). 
We actually look for the case here: `KAFKA=True` + `LOCAL_CLI_RUN=False`.
Your proposal of `self.should_export_plots` would be: `KAFKA=False` + `LOCAL_CLI_RUN=True`
which is indeed the invert of the above.
However, we like to omit sending results for the case: `KAFKA=True` + `LOCAL_CLI_RUN=True`

Reply:
My proposal was to actually use an or condition and not the and condition that you mentioned:
`return self._started_from_cli or not self.kafka_connection.is_enabled()`

Reply:
Ill have an idea how to fix it. Will propose something.